src/utilities/git.jl

```diff
@@ -34,11 +34,11 @@ function git_push(
     ssh = enable_ssh_verbose_b ? "ssh -vvvv" : "ssh"
     git_ssh_command = isnothing(pkey_filename) ? ssh : "$(ssh) -i $pkey_filename"
 
-    withenv("GIT_SSH_COMMAND" => git_ssh_command) do
-        cmd = `git -c user.name="$name" -c user.email="$email" -c committer.name="$name" -c committer.email="$email" push $force_flag $remote $branch`
-        @debug "Attempting to run Git push command" cmd
-        run(cmd)
-    end
+    env2 = copy(ENV);
+    env2["GIT_SSH_COMMAND"] = git_ssh_command
+    cmd = `git -c user.name="$name" -c user.email="$email" -c committer.name="$name" -c committer.email="$email" push $force_flag $remote $branch`
+    @debug "Attempting to run Git push command" cmd env2["GIT_SSH_COMMAND"]
+    run(setenv(cmd, env2))
```

Comment:
Maybe use `addenv`? For instance, something like
```suggestion
    cmd = `git -c user.name="$name" -c user.email="$email" -c committer.name="$name" -c committer.email="$email" push $force_flag $remote $branch`
    @debug "Attempting to run Git push command" cmd git_ssh_command
    run(addenv(cmd, "GIT_SSH_COMMAND" => git_ssh_command))
```